src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponseChatClient.cs

```diff
@@ -0,0 +1,582 @@
+                            result.Tools.Add(ResponseTool.CreateFunctionTool(af.Name, af.Description, functionParameters));
+                            break;
+
+                        case HostedWebSearchTool:
+                            WebSearchToolLocation? location = null;
+                            if (tool.AdditionalProperties.TryGetValue(nameof(WebSearchToolLocation), out object? objLocation))
+                            {
+                                location = objLocation as WebSearchToolLocation;
+                            }
+
+                            WebSearchToolContextSize? size = null;
+                            if (tool.AdditionalProperties.TryGetValue(nameof(WebSearchToolContextSize), out object? objSize) &&
+                                objSize is WebSearchToolContextSize)
+                            {
+                                size = (WebSearchToolContextSize)objSize;
+                            }
+
+                            result.Tools.Add(ResponseTool.CreateWebSearchTool(location, size));
+                            break;
+                    }
+                }
+
+                switch (options.ToolMode)
+                {
+                    case NoneChatToolMode:
+                        result.ToolChoice = _none;
+                        break;
+
+                    case AutoChatToolMode:
+                    case null:
+                        result.ToolChoice = _auto;
+                        break;
+
+                    case RequiredChatToolMode required:
+                        result.ToolChoice = required.RequiredFunctionName is not null ?
+                            BinaryData.FromString($$"""{"type":"function","name":"{{required.RequiredFunctionName}}"}""") :
+                            _required;
+                        break;
+                }
+            }
+
+            // Handle response format.
+            if (options.ResponseFormat is ChatResponseFormatText)
+            {
+                result.TextOptions.ResponseFormat = ResponseTextFormat.CreateTextFormat();
+            }
+            else if (options.ResponseFormat is ChatResponseFormatJson jsonFormat)
+            {
+                result.TextOptions.ResponseFormat = jsonFormat.Schema is { } jsonSchema ?
+                    ResponseTextFormat.CreateJsonSchemaFormat(
+                        jsonFormat.SchemaName ?? "json_schema",
+                        BinaryData.FromBytes(JsonSerializer.SerializeToUtf8Bytes(jsonSchema, OpenAIJsonContext.Default.JsonElement)),
+                        jsonFormat.SchemaDescription) :
+                    ResponseTextFormat.CreateJsonObjectFormat();
+            }
+        }
+
+        return result;
+    }
+
+    /// <summary>Convert a sequence of <see cref="ChatMessage"/>s to <see cref="ResponseItem"/>s.</summary>
+    private static IEnumerable<ResponseItem> ToOpenAIResponseItems(
```

Comment:
The naming of this API creates some incredibly confusing code :) The "response" items are actually the "request" content in this case, because it's a request to the "response" API...

Reply:
Yup. @trrwilson has been trying to figure out how to make that better, but the underlying naming makes it challenging.